save-cli/src/commonMain/kotlin/org/cqfn/save/cli/Config.kt

```diff
@@ -8,72 +8,103 @@ import org.cqfn.save.core.config.LanguageType
 import org.cqfn.save.core.config.ReportType
 import org.cqfn.save.core.config.ResultOutputType
 import org.cqfn.save.core.config.SaveConfig
+import org.cqfn.save.core.logging.logDebug
+import org.cqfn.save.core.logging.logError
 
+import okio.FileSystem
+import okio.IOException
 import okio.Path.Companion.toPath
 
 import kotlinx.cli.ArgParser
 import kotlinx.cli.ArgType
 import kotlinx.cli.default
 import kotlinx.cli.multiple
 import kotlinx.cli.required
+import org.cqfn.save.core.logging.logInfo
+
+private fun <U> Map<String, String>.getAndParseOrElse(
+    key: String,
+    parse: String.() -> U,
+    default: () -> U) =
+        get(key)?.let(parse) ?: default()
 
 /**
  * @param args CLI args
  * @return an instance of [SaveConfig]
+ * @throws e
  */
 @Suppress("TOO_LONG_FUNCTION")
 fun createConfigFromArgs(args: Array<String>): SaveConfig {
-    val parser = ArgParser("save")
+    // The first iteration of parsing - get location of properties file. Ignore any other arguments.
+    val propertiesFile = args.toList().zipWithNext()
+        .firstOrNull { it.first == "--properties-file" || it.first == "-prop" }
+        ?.second
+        ?: "save.properties"
+    logDebug("Using properties file $propertiesFile")
+
+    val properties: Map<String, String> = try {
+        FileSystem.SYSTEM.read(propertiesFile.toPath()) {
+            generateSequence { readUtf8Line() }.toList()
+        }
+            .associate { line ->
+                line.split("=", limit = 2).let {
+                    it.first() to it.last()
+                }
+            }
+    } catch (e: IOException) {
+        logError("Unable to read properties file $propertiesFile: ${e.message}")
+        throw e  // todo exit with exit code
+    }
+    logInfo("Read from properties file: $properties")
+
+    val parser = ArgParser("save")  // todo: or save-cli?
 
     val config by parser.option(
         ArgType.String,
+        fullName = "config",
```

Comment:
Why did you add fullName in this case? Is it redundant, as we already use same name of the `val config`?

Reply:
Yes, you're right. Removed.